src/checkers/inference/solver/backend/z3backend/Z3Backend.java

```diff
@@ -0,0 +1,108 @@
+ package checkers.inference.solver.backend.z3backend;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.AnnotationMirror;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.ErrorReporter;
+
+import com.microsoft.z3.BitVecExpr;
+import com.microsoft.z3.BitVecNum;
+import com.microsoft.z3.BoolExpr;
+import com.microsoft.z3.Context;
+import com.microsoft.z3.Expr;
+import com.microsoft.z3.FuncDecl;
+import com.microsoft.z3.Model;
+import com.microsoft.z3.Optimize;
+
+import checkers.inference.model.Constraint;
+import checkers.inference.model.PreferenceConstraint;
+import checkers.inference.model.Serializer;
+import checkers.inference.model.Slot;
+import checkers.inference.solver.backend.BackEnd;
+import checkers.inference.solver.frontend.Lattice;
+
+public class Z3Backend extends BackEnd<BitVecExpr, BoolExpr>{
```

Comment:
Add Javadoc for class.

Is BitVecExpr the most common format that supports all Z3 SMT theories?

Reply:
> Add Javadoc for class

Will do it.

> Is BitVecExpr the most common format that supports all Z3 SMT theories?

No. I intended leave it as this. Once #65 has been merged, I will adapt Z3 to that refactor and there will not specify encoding types on Backend level anymore.

Reply:
> No. I intend to leave it as this. Once #65 has been merged, I will adapt Z3 to that refactor and there will not specify encoding types on Backend level anymore.

Sounds good.